ioet_feature_flag/toggles.py

```diff
@@ -4,17 +4,17 @@
 
 from . import exceptions, types
 from .providers import Provider, YamlToggleProvider
+from .router import Router
 
 
 _TOGGLES_LOCATION = './feature_toggles/feature-toggles.yaml'
 
 
 class Toggles:
     def __init__(self, provider: typing.Optional[Provider] = None) -> None:
-        if provider:
-            self._provider = provider
-            return
-        self._provider = YamlToggleProvider(_TOGGLES_LOCATION)
+        if not provider:
```

Comment:
I believe that this validation is already been taken into account in the router class